### DISCUSSION TITLE
Naive question re next-urql usage

### DISCUSSION THREAD

**Discussion by lowpitch** [urql-graphql/urql#2136] [Q&A]
Hello!

Apologies in advance for what may be a naive question. We are using `next-urql` and wrapping relevant pages with `withUrqlClient` as required, and SSR working perfectly. 

The site includes a persistent navigation bar that in turn relies on data retrieved from the API - for example, displaying the logged-in user's name / status etc. App structure something like:

`<Layout>
      <Component {...pageProps} />
</Layout>`

...where the `Layout` component in turn contains a `NavBar` component that makes use of `useLoggedInUserQuery` to retrieve user info.

When using `withUrqlClient` to wrap pages, rather than creating and providing a Provider, how best to be using URQL from components that site outside / above pages? 

Calling the query directly from NavBar of course just uses a default URQL client rather than one created using config passed to `withUrlClient`. Wrapping the NavBar component with `withUrqlClient` does sort of work but feels wrong, and fails if SSR-ing with the `ReactDOMServer does not yet support Suspense` error. 

I'm sure I'm missing something obvious here - would be grateful for any pointers. 

TL/DR - when not using a Provider, how to use URQL from within a component that sits outside of a Page?

Thanks in advance.

**Comment by danieledelgiudice** [comment]
I am experiencing the same issue, did you manage to find a solution?
